src/components/RectangleGroup.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useContext, useRef, useEffect } from "react";
+import { Group, Star, Rect, Transformer } from "react-konva";
+import Konva from "konva";
+
+import { CanvasContext } from "../contexts/CanvasContextProvider";
+
+import { Rect as RectClass } from "konva/lib/shapes/Rect";
+import { Transformer as TransformerClass } from "konva/lib/shapes/Transformer";
+import { Rectangle } from "../types";
+import { Group as GroupClass } from "konva/lib/Group";
+
+type RectangleGroupProps = {
+  rect: Rectangle;
+  hasDragEventHandler: (flag: boolean) => void;
+  hasTransformEventHandler: (flag: boolean) => void;
+};
+
+const RectangleGroup = ({
+  rect,
+  hasDragEventHandler,
+  hasTransformEventHandler,
+}: RectangleGroupProps) => {
+  const transformerRef = useRef<TransformerClass | null>(null);
+  const groupRef = useRef<GroupClass | null>(null);
+  const shapeRef = useRef<RectClass | null>(null);
+  // const hasDragEventRef = useRef(false);
+  const {
+    rects,
+    deleteRectangleById,
+    toggleIsHoveredById,
+    toggleIsSelectedById,
+    updateRectangle,
+    updateRectangleList,
+  } = useContext(CanvasContext);
+
+  const dragStartHandler = (event: Konva.KonvaEventObject<DragEvent>) => {
+    hasDragEventHandler(true);
+  };
+
+  const dragEndHandler =
+    (event: Konva.KonvaEventObject<DragEvent>) => (id: string) => {
+      const { clientX, clientY, offsetX, offsetY } = event.evt;
+
+      // console.log("====== drag ends", {
+      //   clientX,
+      //   clientY,
+      //   offsetX,
+      //   offsetY,
+      // });
```

Comment:
Remove unnecessary comments

Reply:
Got it. I was doing some trials and errors so I commented them out just for later use.